### PR TITLE
feat: Add support for template previews

### PR DESCRIPTION
**Pull request by yourtallness**
## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] 🌟 New feature (non-breaking change which adds functionality)
- [ ] 🔨 Breaking change (fix or feature that would cause existing functionality)
- [X] 📖 Docs change / refactoring / dependency upgrade to change)

## Description

Added support for the `POST /v1/templates/{template_type}/{slug}/preview` backend API endpoint.

### CODE REVIEW

**Review by gkats — APPROVED** [review]
Left some comments. Most important, let's handle all errors.

File: clerk/templates_test.go [hunk 1]
```diff
@@ -175,9 +207,16 @@ const dummyTemplateJSON = `{
...
 	"body": "<p>One quarter of a mile at a time<p>"
 }`
+
+const dummyPreviewRequestJSON = `{
+	"body": "{{OTPCode}} is your code for {{AppName}}, valid for {{TTLMinutes}} minutes"
+}`
+
+const dummyPreviewResponseJSON = `{
+	"body": "123456 is your code for ACME, valid for 10 minutes"
+}`
```
[the remark this answers is not in the document]
Comment (yourtallness) [reply clerk/templates_test.go:222]: What do you recommend?

File: clerk/templates.go [hunk 2]
```diff
@@ -102,3 +112,18 @@ func (s *TemplatesService) Delete(templateType, slug string) (*DeleteResponse, e
 
 	return &delResponse, nil
 }
+
+// Preview returns a rendering of a template with sample data for preview purposes
+func (s *TemplatesService) Preview(templateType, slug string, previewTemplateRequest *PreviewTemplateRequest) (*TemplatePreview, error) {
+	templateURL := fmt.Sprintf("%s/%s/%s/preview", TemplatesUrl, templateType, slug)
```
[the remark this answers is not in the document]
Comment (yourtallness) [reply clerk/templates.go:118]: Yeah, the sdk could use a URL builder, but I'd handle this in another PR.

File: clerk/templates.go [hunk 3]
```diff
@@ -102,3 +112,18 @@ func (s *TemplatesService) Delete(templateType, slug string) (*DeleteResponse, e
...
+	req, _ := s.client.NewRequest("POST", templateURL, previewTemplateRequest)
```
[the remark this answers is not in the document]
Comment (yourtallness) [reply clerk/templates.go:119]: True, I'm doing it in a manner consistent with the rest of the sdk.

I'll open a task for this.

File: clerk/templates_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gkats) [reply clerk/templates_test.go:222]: Just add them inside the test, inline. No biggie if you don't want to do it of course.

File: clerk/templates.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (gkats) [reply clerk/templates.go:119]: :+1: Wow, I'm surprised to hear that we ignore all these errors when creating request objects.

**Pull request merged**